lifetimes/fitters/beta_geo_fitter.py

```diff
@@ -234,24 +234,31 @@ def conditional_expected_number_of_purchases_up_to_time(
 
         r, alpha, a, b = self._unload_params("r", "alpha", "a", "b")
 
-        numerator = (
-            1
-            - ((alpha + T) / (alpha + T + t)) ** (r + frequency)
-            * hyp2f1(
-                r + frequency,
-                b + frequency,
-                a + b + frequency - 1,
-                t / (alpha + T + t),
-            )
+        _a = r + frequency
+        _b = b + frequency
+        _c = a + b + frequency - 1
+        _z = t / (alpha + T + t)
+
+        hyp_term = hyp2f1(_a, _b, _c, _z)
+        first_term = (a + b + frequency - 1) / (a - 1)
+        second_term = 1 - ((alpha + T) / (alpha + T + t)) ** (r + frequency) * hyp_term
```

Comment:
Why are you keeping the non-alt version?  @pavlo-seimskyi

Reply:
It is used if the result of `hyp2f1` is not `inf`

Reply:
But why isn't it always used? What's the benefit of using the other when is not infinite?